de-lib/src/main/java/org/iplantc/de/client/models/UserSettings.java

```diff
@@ -3,68 +3,36 @@
+import org.iplantc.de.client.models.userSettings.UserSetting;
+import org.iplantc.de.client.models.userSettings.UserSettingAutoBeanFactory;
 
+import com.google.common.base.Strings;
 import com.google.gwt.core.client.GWT;
-import com.google.gwt.json.client.JSONObject;
-import com.google.web.bindery.autobean.shared.AutoBean;
-import com.google.web.bindery.autobean.shared.AutoBeanCodex;
-import com.google.web.bindery.autobean.shared.AutoBeanUtils;
-import com.google.web.bindery.autobean.shared.Splittable;
-import com.google.web.bindery.autobean.shared.impl.StringQuoter;
 
 /**
- * 
  * A singleton hold user general settings
- * 
+ *
  * @author sriram
- * 
  */
 public class UserSettings {
 
     private final KeyBoardShortcutConstants SHORTCUTS = GWT.create(KeyBoardShortcutConstants.class);
-    private boolean enableAnalysisEmailNotification;
-    private boolean enableImportEmailNotification;
-    private String defaultFileSelectorPath;
-    private boolean rememberLastPath;
-    private boolean saveSession;
-    private boolean userSessionConnection;
-    private Folder defaultOutputFolder;
-    private String dataShortCut;
-    private String appShortCut;
-    private String analysesShortCut;
-    private String notifyShortCut;
-    private String closeShortCut;
-    private Folder systemDefaultOutputFolder;
-    private String lastPath;
-    private boolean enableWaitTimeMessage;
-    
-    
-    public static final String EMAIL_ANALYSIS_NOTIFICATION = "enableAnalysisEmailNotification";
-    public static final String EMAIL_IMPORT_NOTIFICATION = "enableImportEmailNotification";
-    public static final String DEFAULT_FILE_SELECTOR_PATH = "defaultFileSelectorPath";
-    public static final String REMEMBER_LAST_PATH = "rememberLastPath";
-    public static final String SAVE_SESSION = "saveSession";
-    public static final String DEFAULT_OUTPUT_FOLDER = "defaultOutputFolder";
-    public static final String DATA_KB_SHORTCUT = "dataKBShortcut";
-    public static final String APPS_KB_SHORTCUT = "appsKBShortcut";
-    public static final String ANALYSIS_KB_SHORTCUT = "analysisKBShortcut";
-    public static final String NOTIFICATION_KB_SHORTCUT = "notificationKBShortcut";
-    public static final String CLOSE_KB_SHORTCUT_STRING = "closeKBShortcut";
-    public static final String SYSTEM_DEFAULT_OUTPUT_DIR = "systemDefaultOutputDir";
-    public static final String LAST_PATH = "lastPathId";
-    public static final String WAIT_TIME_MESSAGE = "enableWaitTimeMessage";
-
+    private final DiskResourceAutoBeanFactory diskResourceFactory =
+            GWT.create(DiskResourceAutoBeanFactory.class);
+    private final UserSettingAutoBeanFactory factory = GWT.create(UserSettingAutoBeanFactory.class);
+    private UserInfo userInfo;
 
+    private UserSetting userSetting;
     private static UserSettings instance;
+    private boolean userSessionConnection;
+    private static String ANALYSES_DIR = "analyses";
```

Comment:
This should probably come from the configs, but that doesn't need to happen in this PR, so we can get this merged this for the next QA drop 😁